pymc/model_graph.py

```diff
@@ -302,33 +335,50 @@ def get_plates(self, var_names: Iterable[VarName] | None = None) -> dict[str, se
         """
         plates = defaultdict(set)
 
-        # TODO: Evaluate all RV shapes and dim_length at once.
-        #       This should help to find discrepancies, and
-        #       avoids unnecessary function compiles for deetermining labels.
+        # TODO: Evaluate all RV shapes at once
+        #       This should help find discrepencies, and
+        #       avoids unnecessary function compiles for determining labels.
+        dim_lengths: dict[str, int] = {
+            name: fast_eval(value).item() for name, value in self.model.dim_lengths.items()
+        }
 
         for var_name in self.vars_to_plot(var_names):
             v = self.model[var_name]
-            shape: Sequence[int] = fast_eval(v.shape)
-            dim_labels = []
+            shape: tuple[int, ...] = tuple(fast_eval(v.shape))
             if var_name in self.model.named_vars_to_dims:
                 # The RV is associated with `dims` information.
+                names = []
+                sizes = []
                 for d, dname in enumerate(self.model.named_vars_to_dims[var_name]):
-                    if dname is None:
-                        # Unnamed dimension in a `dims` tuple!
-                        dlen = shape[d]
-                        dname = f"{var_name}_dim{d}"
-                    else:
-                        dlen = fast_eval(self.model.dim_lengths[dname])
-                    dim_labels.append(f"{dname} ({dlen})")
-                plate_label = " x ".join(dim_labels)
+                    names.append(dname)
+                    sizes.append(dim_lengths.get(dname, shape[d]))
+
+                plate_meta = PlateMeta(
+                    names=tuple(names),
+                    sizes=tuple(sizes),
+                )
             else:
                 # The RV has no `dims` information.
-                dim_labels = [str(x) for x in shape]
-                plate_label = " x ".join(map(str, shape))
-            plates[plate_label].add(var_name)
+                sizes = tuple(shape)
+                plate_meta = PlateMeta(
+                    names=tuple([var_name] * len(sizes)),
+                    sizes=sizes,
+                    dim_info=False,
+                )
+
+            v = self.model[var_name]
+            node_type = get_node_type(var_name, self.model)
+            var = NodeMeta(var=v, node_type=node_type)
+            plates[plate_meta].add(var)
 
         return dict(plates)
 
+    def edges(self, var_names: Iterable[VarName] | None = None):
+        for child, parents in self.make_compute_graph(var_names=var_names).items():
+            # parents is a set of rv names that precede child rv nodes
+            for parent in parents:
+                yield child.replace(":", "&"), parent.replace(":", "&")
+
     def make_graph(
```

Comment:
Should `make_graph` and `make_networkx` now be functions that take plates and edges as inputs?

Reply:
That would just remove calling `get_plates` and `edges` methods. Don't have much of a preference

Reply:
It would make it more modular, in that if you find a way to create your own plates and edges, you can just pass it to the functions that then display it?

Reply:
Yeah, sure. I think that makes sense then. 

The dictionary of {PlateMeta : set[NodeMeta]} is a bit weird and hard to work with. i.e. set is not subscritable and looking up by PlateMeta key is a bit tricky.

I was thinking of having another object, `Plate` which would be:

```python
@dataclass 
class Plate: 
    plate_meta: PlateMeta
    nodes: list[NodeMeta]
```
and that would be in the input to `make_graph` and `make_networkx` instead. Making the signature: (plates: list[Plate], edges: list[tuple[str, str]], ...)

Reply:
Also, does it make sense as a method still? Do you see model_to_graphviz taking this input as well?

Reply:
Lost track of the specific methods we're discussing. My low resolution guess was that once we have the plates / edges we can just pass them to a function that uses those to render graphviz or networkx graphs. Let me know if you were asking about something else or see a problem (or no point) with that

Reply:
Sounds good. Let me push something up and you can give feedback

Reply:
Just pushed. 
If user has arbitrary list[Plate] and list[tuple[VarName, VarName]] then they can use `make_graph` or `make_networkx` in order to make the graphviz or networkx, respectively. 
`pm.model_to_graphviz` and `pm.model_to_networkx` are still wrappers. 
`ModelGraph` class can be used to create the plates and edges in the previous manner if desired with the `get_plates` and `edges` methods